core/frontend/src/internal/render/webgl/SurfaceGeometry.ts

```diff
@@ -36,12 +40,73 @@ function wantLighting(vf: ViewFlags) {
 export class SurfaceGeometry extends MeshGeometry {
   private readonly _buffers: BuffersContainer;
   private readonly _indices: BufferHandle;
+  public readonly hasTextures: boolean;
+  public textureParams: LayerTextureParams | undefined;
 
   public get lutBuffers() { return this._buffers; }
 
-  public static create(mesh: MeshData, indices: VertexIndices): SurfaceGeometry | undefined {
+  public static create(mesh: MeshData, params: MeshParams): SurfaceGeometry | undefined {
+    const indices = params.surface.indices;
     const indexBuffer = BufferHandle.createArrayBuffer(indices.data);
-    return undefined !== indexBuffer ? new SurfaceGeometry(indexBuffer, indices.length, mesh) : undefined;
+
+    const tile = params.tile;
+
+    const layerClassifiers = tile?.tree?.layerHandler?.layerClassifiers;
+
+    if (!layerClassifiers?.size || !tile || undefined === layerClassifiers) return undefined !== indexBuffer ? new SurfaceGeometry(indexBuffer, indices.length, mesh, undefined) : undefined;
+
+    const transformECEF = tile.tree.iModel.getEcefTransform();
+    if (!transformECEF || !tile.range) return undefined !== indexBuffer ? new SurfaceGeometry(indexBuffer, indices.length, mesh, undefined) : undefined;
```

Comment:
```suggestion
    if (!layerClassifiers?.size || !tile || undefined === layerClassifiers) {
      return undefined !== indexBuffer ? new SurfaceGeometry(indexBuffer, indices.length, mesh, undefined) : undefined;
    }

    const transformECEF = tile.tree.iModel.getEcefTransform();
    if (!transformECEF || !tile.range) {
      return undefined !== indexBuffer ? new SurfaceGeometry(indexBuffer, indices.length, mesh, undefined) : undefined;
    }
```
I hate nitpicking on whitespace (we should be using an automated code formatter) but please keep readability in mind.